### DISCUSSION TITLE
rust-course/关于生命周期的理解

### DISCUSSION THREAD

**Discussion by OuYangPaste** [sunface/rust-course#339] [内容建议]
不太明白为什么现在很多关于生命周期的讲解都喜欢从把 'a 的范围推断为 所有标记 'a 的入参的存活交集范围, 我个人来说习惯把 返回接收的引用所存活的范围定性为 'a ，而入参的范围定性为 'b: 'a, 'c: 'a，特别在 rust 有了 NLL 之后，借用检查器对变量的生命周期检查粒度更细了，而生命周期的本质是帮编译器把 返回的引用 和 被引用的资源 重新挪回到借用检查这个体系当中
    （emmm 用手机写的不太方便举例子😂, 希望上面的文字能表达清楚我的意思)

**Comment by sunface** [comment]
因为 'a 确实是一个交集 = ， =，NLL属于额外的规则，不应该跟已有的规则混淆在一起，未来可能会有其他的规则，甚至不再有NLL :)

**Reply by OuYangPaste** [reply]
> 因为 'a 确实是一个交集 = ， =，NLL属于额外的规则，不应该跟已有的规则混淆在一起，未来可能会有其他的规则，甚至不再有NLL :)

但是从交集的角度上来想，返回的引用的生命周期和最短生命周期入参同为 'a，也意味着这个最短生命周期的入参将会被借用到释放为止?
NLL 可能会被替换，但是生命周期的检查应该只会变得更只能，粒度更细
